### ISSUE TITLE
Implement partial transformations

### ISSUE THREAD

**Issue by james-d-mitchell** [semigroups/Semigroups#144]
Originally reported by: **James Mitchell (Bitbucket: [james-d-mitchell](http://bitbucket.org/james-d-mitchell), GitHub: [james-d-mitchell](http://github.com/james-d-mitchell))**

---

Essentially make a wrapper for transformations, which implement partial transformations. Requested by Itamar Stein.

---
- Bitbucket: https://bitbucket.org/james-d-mitchell/semigroups/issue/145

**Comment by wilfwilson** [comment]
@james-d-mitchell Could you give some more information about how you propose this would be solved?

**Comment by james-d-mitchell** [comment]
I had in mind to simply wrap transformations and semigroups of transformations, so that they appear to be partial transformations. Several people requested this feature but I don't think it makes much sense to do anything other than what I just wrote. 

On the other hand, I'd like to finish the next major release before adding more features.

**Comment by wilfwilson** [comment]
Yes, I wasn't proposing that I or anyone else starts work on this. I was just looking through the GitHub issues and thinking about them in turn. For this one I don't see immediately how I'd go about doing the wrapping. I'll talk about it with you sometime.
